src/Microsoft.AspNetCore.Blazor/Json/CamelCase.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Blazor.Json
+{
+    internal static class CamelCase
+    {
+        public static string ToCamelCase(this string value)
+        {
+            // Basic cases: if we don't need to modify the value, bail out without creating a char array
+            if (string.IsNullOrEmpty(value))
+            {
+                return value;
+            }
+
+            if (!char.IsUpper(value[0]))
+            {
+                return value;
+            }
+
+            // We have to modify at least one character
+            var chars = value.ToCharArray();
+
+            var length = chars.Length;
+            if (length < 2 || !char.IsUpper(chars[1]))
+            {
+                // Basic cases where only the first char needs to be modified
+                chars[0] = char.ToLowerInvariant(chars[0]);
+            }
```

Comment:
Is there much reason to have this special case? It looks to me like this would immediately hit the 'break' condition in the loop.

Reply:
Ah yes, I see it's not clear from the comment but this branch is functionally required. Without it, the logic would never look at the value of `chars[1]` and would produce an incorrect result for inputs like `OnCLICK` (it would produce `onclick` instead of `onCLICK`).

Basically `chars[0]` and `chars[1]` are both special, because the "when to stop" logic is based on looking at the `n+2` character versus the one we're currently lowercasing. Since the loop starts at index 0 that means it only starts looking at the case of characters from 2 onwards.

I'll update the comment to clarify this.

Reply:
Done